### ISSUE TITLE
8.2.4.1 creates invalid file list for certain rsyslog configs

### ISSUE THREAD

**Issue by systeminsightsbuild** [awailly/cis-ubuntu-ansible#86]
I have the config

```
$ cat /etc/rsyslog.d/30-marathon.conf
if $programname == 'marathon' then {
  action(type="omfile" asyncWriting="on" file="/var/log/marathon.log")
  stop
}
```

This puts the value `file="/var/log/marathon.log"` in `results`, which ends up failing 8.2.4.3 since that is not a simple file name.

I'm a bit of novice when it comes to syslog configs -- I just copied the above from somewhere and made it part of my marathon install role. 

Is this is case that should be handled? Or should I have a different form in the config?

**Comment by systeminsightsbuild** [comment]
I cobbled together a sed that cleaned up the filename. Not sure if this is best way to handle this though.

current file list:

```
/var/log/cloud-init.log
/var/log/syslog
file="/var/log/marathon.log"
/var/log/auth.log
/var/log/syslog
```

Using:

```
awk '{ print $NF }' /etc/rsyslog.d/* /etc/rsyslog.conf | grep /var/log | sed 's/^-//' | sed 's/)$//' | sed 's/^file=\"\(.*\)\"/\1/'
```

cleaned file list:

```
/var/log/cloud-init.log
/var/log/syslog
/var/log/marathon.log
/var/log/auth.log
/var/log/syslog
```

This seems so ad-hoc.

**Comment by awailly** [comment]
Ok this one needs more attention, I was not aware of advanced rsyslog configurations.

**Comment by memelet** [comment]
Any thoughts on what to do with this?

**Comment by awailly** [comment]
I tried to get the maximum number of files that rsyslog can write to, but there is room for user defined scripts and I unfortunately cannot see any command that we can run to catch all of them.

**Comment by pchaigno** [comment]
What if we use one the following regexp instead of the one in 8.2.4.1:

``` shell
awk '{ print $NF }' /etc/rsyslog.d/* /etc/rsyslog.conf | grep -ioP "/var/log/[\w-]+(\.log)?"
awk '{ print $NF }' /etc/rsyslog.d/* /etc/rsyslog.conf | grep -ioP "/var/log/[\w-\.]+"
```

It does suppose log files are in `/var/log` and their names only contain alphanumeric characters, but that seems like a pretty fair assumption, no?

**Comment by memelet** [comment]
@pchaigno I like that.

**Comment by pchaigno** [comment]
Fixed in #91.
